amgut/db/patches/0009.sql

```diff
@@ -0,0 +1,29 @@
+-- Add created_on timestamp for handout kits
+ALTER TABLE ag.ag_handout_kits ADD created_on timestamp DEFAULT current_timestamp NOT NULL;
+
+-- Create new ag_handout_barcodes table
+CREATE TABLE ag.ag_handout_barcodes (
+	kit_id               varchar  NOT NULL,
+	barcode              varchar(9)  NOT NULL,
+	sample_barcode_file  varchar(13),
+	CONSTRAINT idx_ag_handout_barcodes PRIMARY KEY ( barcode )
+ );
+
+CREATE INDEX idx_ag_handout_barcodes_0 ON ag.ag_handout_barcodes ( barcode );
+
+CREATE INDEX idx_ag_handout_barcodes_1 ON ag.ag_handout_barcodes ( kit_id );
+
+ALTER TABLE ag.ag_handout_barcodes ADD CONSTRAINT fk_ag_handout_barcodes FOREIGN KEY ( barcode ) REFERENCES ag.barcode( barcode );
+
+-- Copy needed data to new table
+INSERT INTO ag.ag_handout_barcodes (kit_id, barcode, sample_barcode_file)
+SELECT kit_id, barcode, sample_barcode_file FROM ag.ag_handout_kits;
+
+-- Delete duplicate handout rows and unneeded columns from handouts table
+DELETE FROM ag.ag_handout_kits WHERE barcode NOT IN (SELECT max(barcode) FROM ag.ag_handout_kits);
```

Comment:
Josh, can you please verify this with a local copy of the production data?

On Mon, Jul 27, 2015 at 10:05 AM, Joshua Shorenstein <
notifications@github.com> wrote:

> In amgut/db/patches/0009.sql
> <https://github.com/biocore/american-gut-web/pull/428#discussion_r35553447>
> :
>
> > +	sample_barcode_file  varchar(13),
> > +	CONSTRAINT idx_ag_handout_barcodes PRIMARY KEY ( barcode )
> > + );
> > +
> > +CREATE INDEX idx_ag_handout_barcodes_0 ON ag.ag_handout_barcodes ( barcode );
> > +
> > +CREATE INDEX idx_ag_handout_barcodes_1 ON ag.ag_handout_barcodes ( kit_id );
> > +
> > +ALTER TABLE ag.ag_handout_barcodes ADD CONSTRAINT fk_ag_handout_barcodes FOREIGN KEY ( barcode ) REFERENCES ag.barcode( barcode );
> > +
> > +-- Copy needed data to new table
> > +INSERT INTO ag.ag_handout_barcodes (kit_id, barcode, sample_barcode_file)
> > +SELECT kit_id, barcode, sample_barcode_file FROM ag.ag_handout_kits;
> > +
> > +-- Delete duplicate handout rows and unneeded columns from handouts table
> > +DELETE FROM ag.ag_handout_kits WHERE barcode NOT IN (SELECT max(barcode) FROM ag.ag_handout_kits);
>
> Oh wow, that is special. Thought I had the groupby in there [image:
> :disappointed:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/american-gut-web/pull/428/files#r35553447>.
>